Add the 9.0.0 changelog. (#20090)
coq/coq: doc/sphinx/changes.rst

  :n:`Derive open_binders in type as name`, so that several names,
  and possibly types to these names, can be given
  (`#19295 <https://github.com/coq/coq/pull/19295>`_,
  by Hugo Herbelin).
- **Changed:**
  `match` elaboration can unify sort quality variables to make an elimination valid
  (`#19329 <https://github.com/coq/coq/pull/19329>`_,
  fixes `#19327 <https://github.com/coq/coq/issues/19327>`_,
  by Gaëtan Gilbert).
- **Changed:**
  ``:>`` in :token:`of_type_inst` now always declares coercions. The previous behavior,
  deprecated since 8.18, was to declare typeclass instances instead,
  when used in records declared with the :cmd:`Class` keyword. Look at
  the :ref:`previous changelog entries <819_changes_spec_language>`
  about former warnings `future-coercion-class-constructor` and
  `future-coercion-class-field` for advice on how to update your code
  (`#19519 <https://github.com/coq/coq/pull/19519>`_, by Pierre Roux).
- **Changed:**
  The unification algorithm does not solve unification problems of the
  form `proj _ ~ _` using canonical structures when the LHS reduces or
  is ground
  (`#19611 <https://github.com/coq/coq/pull/19611>`_,

>   by ).

Reviewer: @Tragicus shouldn't this read `by Quentin Vermande`?

- **Fixed:**
  :cmd:`Derive` now supports :cmd:`Admitted`
  (`#19092 <https://github.com/coq/coq/pull/19092>`_,
  fixes `#18951 <https://github.com/coq/coq/issues/18951>`_,
  by Hugo Herbelin).
- **Fixed:**
  Mishandling of let binders in `Program Fixpoint`
  (`#19257 <https://github.com/coq/coq/pull/19257>`_,
  fixes `#16906 <https://github.com/coq/coq/issues/16906>`_,
  by Hugo Herbelin).
- **Fixed:**
  Pattern-matching in :attr:`Program` mode now supports inductive
  types using :ref:`local definitions <let-in>` in their declaration
  (`#19773 <https://github.com/coq/coq/pull/19773>`_,
  fixes `#10407 <https://github.com/coq/coq/issues/10407>`_,
  by Hugo Herbelin).
- **Fixed:**
  Anomaly in :cmd:`Function` when a well-founded relation had not the expected type
  (`#19775 <https://github.com/coq/coq/pull/19775>`_,
  fixes `#12417 <https://github.com/coq/coq/issues/12417>`_,
  by Hugo Herbelin).

Notations
^^^^^^^^^

- **Fixed:**
  Recognized all Unicode non-spacing marks as valid identifier characters
  (`#19693 <https://github.com/coq/coq/pull/19693>`_,
  fixes `#19512 <https://github.com/coq/coq/issues/19512>`_,
  by Guillaume Melquiond).

Tactics
^^^^^^^

- **Changed:**

> 

Reviewer: ```suggestion
```

  scrupulously :g:`simpl never` and to stop reducing further in
  subterms that are *not* in position of argument of a destructor, as
  specified by :g:`simpl never`
  (`#18591 <https://github.com/coq/coq/pull/18591>`_,
  fixes `#16040 <https://github.com/coq/coq/issues/16040>`_,
  by Hugo Herbelin).
- **Fixed:**
  `Set Typeclasses Strict Resolution` is no longer ignored in
  `typeclasses eauto with <dbs>`
  (`#19436 <https://github.com/coq/coq/pull/19436>`_,
  fixes `#15432 <https://github.com/coq/coq/issues/15432>`_,
  by Jan-Oliver Kaiser).
- **Fixed:**
  Unbound variables were sometimes generated when a metavariable of a
  theorem given to :tacn:`apply` occurred in the type of the theorem
  under a :n:`fun`
  (`#19769 <https://github.com/coq/coq/pull/19769>`_,
  fixes `#17314 <https://github.com/coq/coq/issues/17314>`_,
  by Hugo Herbelin).
- **Fixed:**
  `cbn` now considers primitive literals (integers, floats, arrays, strings)
  "constructors", i.e. they now satisfy the `!` modifier in `Arguments`
  (`#20004 <https://github.com/coq/coq/pull/20004>`_,
  fixes `#20003 <https://github.com/coq/coq/issues/20003>`_,
  by Jan-Oliver Kaiser).

> 

Reviewer: ```suggestion
```

Ltac2 language
^^^^^^^^^^^^^^

> - **Added:**

Reviewer: ```suggestion

- **Added:**
```

  Added Ltac2 bindings for congruence and simpl congruence, it fixes #14289 not entirely but provides Ltac2 bindings for one of the tactics listed there
  (`#19032 <https://github.com/coq/coq/pull/19032>`_,
  fixes `#14289 <https://github.com/coq/coq/issues/14289>`_,

>   by Benny Smit, review of Jason Gross, review of Pierre-Marie Pédrot, review of Gaëtan Gilbert).

Reviewer: ```suggestion
  by Benny Smit, reviewed by Jason Gross, Pierre-Marie Pédrot and Gaëtan Gilbert).
```

  (`#19575 <https://github.com/coq/coq/pull/19575>`_,
  by Gaëtan Gilbert).
- **Added:**
  `Ltac2.Constr.noccur_between` and `noccurn` to test for non-occurrence of local variables in terms
  (`#19614 <https://github.com/coq/coq/pull/19614>`_,
  by Gaëtan Gilbert).
- **Deprecated:**
  `Ltac2.Constr.occur_between` and `occurn` whose return values are the opposite of that implied by their names
  (`#19614 <https://github.com/coq/coq/pull/19614>`_,
  by Gaëtan Gilbert).
- **Added:**
  `Ltac2.Control.hyp_value` to get the value (`v` in `H := v`) of an hypothesis
  (`#19630 <https://github.com/coq/coq/pull/19630>`_,
  by Gaëtan Gilbert).
- **Fixed:**
  resolution of :ref:`abbreviations <Abbreviations>` in :n:`reference`
  in :token:`ltac2_quotations`, for instance in eval tactic
  delta-reduction flags :token:`ltac2_delta_reductions`
  (`#19589 <https://github.com/coq/coq/pull/19589>`_,
  fixes `#19590 <https://github.com/coq/coq/issues/19590>`_,
  by Pierre Roux).
- **Fixed:**
  `Ltac2 Eval` does not require to be focussed in a goal
  anymore (`#19961 <https://github.com/coq/coq/pull/19961>`_, by
  Daniil Iaitskov).

SSReflect
^^^^^^^^^

> - **Changed:**
>   The :tacn:`done` tactic now tries to apply `sym_equal` with four arguments
>   instead of trying first with zero to three arguments.

Reviewer: I guess
```suggestion
- **Changed:**
  The :tacn:`done` tactic now tries to apply `sym_equal` with four arguments
  instead of trying first with zero to three arguments
  (`#19372 <https://github.com/coq/coq/pull/19372>`_,
  by Quentin Vermande).
```
@Tragicus do you confirm?

> - **Changed:**
>   `done` uses `simple refine` instead of `apply` to apply `sym_equal`

Reviewer: I guess
```suggestion
- **Changed:**
  `done` uses `simple refine` instead of `apply` to apply `sym_equal`
  (`#19399 <https://github.com/coq/coq/pull/19399>`_,
  by Quentin Vermande).
```
@Tragicus do you confirm?

  doesn't exist. This enables easier use of the compat mechanism with
  versions where the compatibility file doesn't exist yet
  (`#19370 <https://github.com/coq/coq/pull/19370>`_,
  by Pierre Roux).
- **Changed:**
  `coq_makefile` generated makefiles only install plugin `.cmxs` files in findlib locations
  and stop putting a copy in `user-contrib` (the copy should be useless after the removal of plugin legacy loading)
  (`#19841 <https://github.com/coq/coq/pull/19841>`_,
  by Gaëtan Gilbert).
- **Removed:**
  `coqdep` flag `-m` (it was used through `coq_makefile`)
  (`#19863 <https://github.com/coq/coq/pull/19863>`_,
  by Gaëtan Gilbert).
- **Added:**
  :ref:`command line option <command-line-options>` ``-compat-from``
  to enable writing compatibility files for libraries similarly to the
  ``-compat`` option for Rocq
  (`#19370 <https://github.com/coq/coq/pull/19370>`_,
  by Pierre Roux).
- **Added:**
  The ``-compat`` :ref:`command line option <command-line-options>`
  now silences deprecation warnings that were introduced since the
  given version
  (`#19370 <https://github.com/coq/coq/pull/19370>`_,
  by Pierre Roux).

> RocqIDE
> ^^^^^^^
> - **Changed:**

Reviewer: ```suggestion
RocqIDE
^^^^^^^

- **Changed:**
```

  rather than immediately
  (`#19417 <https://github.com/coq/coq/pull/19417>`_,
  by Sylvain Chiron).
- **Fixed:**
  Changing the allowed modifiers in the Shortcuts panel of
  the Preferences dialog now immediately updates the available
  modifiers for the listed items
  (`#19417 <https://github.com/coq/coq/pull/19417>`_,
  by Sylvain Chiron).
- **Added:**
  Preference setting for unjustified conclusions background color
  (`#19417 <https://github.com/coq/coq/pull/19417>`_,
  by Sylvain Chiron).
- **Changed:**
  CoqIDE is renamed to RocqIDE (the auxiliary binary `coqidetop` is not renamed)
  (`#20036 <https://github.com/coq/coq/pull/20036>`_,
  by Gaëtan Gilbert).
- **Added:**
  Warnings are now included in the Errors panel
  (`#19188 <https://github.com/coq/coq/pull/19188>`_,
  by Jim Fehrle).
- **Fixed:**
  Changing the position of buffer names (top, left,
  bottom or right) no longer needs a restart
  (`#19166 <https://github.com/coq/coq/pull/19166>`_,
  by Sylvain Chiron).
- **Added:**
  Document tabs are now reorderable
  (`#19166 <https://github.com/coq/coq/pull/19166>`_,
  by Sylvain Chiron).

Standard library
^^^^^^^^^^^^^^^^

- **Changed:**

>   Stdlib moved to its own repository, look for Stdlib own changelog

Reviewer: ```suggestion
  Stdlib moved to its own repository, look for Stdlib own changelog
```
I guess we should add a link to refman-stdlib here @Zimmi48 or @mattam82 would you know what the exact link would be with the new website?

Reviewer: ```suggestion
  Stdlib moved to its own repository, look for `Stdlib own changelog <https://rocq-prover.org/doc/v9.0/refman-stdlib/changes.html>`_
```

  for other changes there
  (`#19975 <https://github.com/coq/coq/pull/19686>`_,
  by Pierre Roux).
- **Added:**

>   a new `rocq-core` package for users who don't want to depend on Stdlib

Reviewer: Maybe (but again, not sure what the exact link should be)
```suggestion
  a new `rocq-core` package for users who don't want to depend on Stdlib.
  This packages provides `Corelib` <link to corelib doc>
```

Reviewer: ```suggestion
  a new `rocq-core` package for users who don't want to depend on Stdlib.
  This provides `Corelib <https://rocq-prover.org/doc/v9.0/corelib>`_ a tiny subset of Stdlib
```
